includes/Modules/Search_Console.php

```diff
@@ -360,57 +347,41 @@ protected function create_search_analytics_data_request( array $args = array() )
 	}
 
 	/**
-	 * Checks whether Search Console data exists for the given post.
+	 * Checks whether Search Console data exists for the given URL.
 	 *
 	 * The result of this query is stored in a transient.
 	 *
 	 * @since 1.0.0
 	 *
-	 * @param string $current_url The current url.
+	 * @param string $url The url to check data for.
 	 * @return bool True if Search Console data exists, false otherwise.
 	 */
-	protected function has_data_for_url( $current_url ) {
-		if ( ! $current_url ) {
+	protected function has_data_for_url( $url ) {
+		if ( ! $url ) {
 			return false;
 		}
 
-		$transient_key = 'googlesitekit_sc_data_' . md5( $current_url );
+		$transient_key = 'googlesitekit_sc_data_' . md5( $url );
 		$has_data      = get_transient( $transient_key );
-		if ( false === $has_data ) {
 
-			// Check search console for data.
-			$datasets = array(
+		if ( false === $has_data ) {
+			/* @var Google_Service_Webmasters_ApiDataRow[]|WP_Error $response_rows Array of data rows. */
+			$response_rows = $this->get_data(
+				'searchanalytics',
 				array(
-					'identifier' => $this->slug,
-					'key'        => 'sc-site-analytics',
-					'datapoint'  => 'searchanalytics',
-					'data'       => array(
-						'url'               => $current_url,
-						'dateRange'         => 'last-90-days',
-						'dimensions'        => 'date',
-						'compareDateRanges' => true,
-					),
-				),
-			);
-
-			$responses = $this->get_batch_data(
-				array_map(
-					function( $dataset ) {
-						return (object) $dataset;
-					},
-					$datasets
+					'url'               => $url,
+					'dimensions'        => 'date',
+					'compareDateRanges' => true,
 				)
 			);
 
-			$has_data = false;
-
-			// Go thru results, any impressions means the URL has data.
-			foreach ( $responses['sc-site-analytics'] as $key => $response ) {
-				if ( is_wp_error( $response ) || empty( $response ) || ! isset( $response ) ) {
-					continue;
-				}
+			if ( is_wp_error( $response_rows ) ) {
+				return false;
+			}
```

Comment:
See above, this should remain being cached as `0` for an hour like before.

Reply:
Good catch 👍